### ISSUE TITLE
questions about l_inf norm calculation

### ISSUE THREAD

**Issue by lihenglin** [henrygouk/mars-finetuning#3]
Hi,

I am confused about the calculation of l_inf norm in the lipschitz.py file.
I cannot understand why the K.sum operation (line 87) is running through the axis 0 for dense layer.
From my point of view, I think it should run through the axis 1 such that the obtained tensor can be of the shape (r, 1).
Do I miss anything here?

Thanks.

**Comment by lihenglin** [comment]
Never mind. I realized that the dimensions of the weight matrices is (input_dim, output_dim) in tensorflow, so your implementation is right. Sorry for the disturbance.
